packages/apollo-server-core/src/requestPipeline.ts

```diff
@@ -290,26 +290,47 @@ export async function processGraphQLRequest<TContext>(
       );
     }
 
-    const executionDidEnd = await dispatcher.invokeDidStartHook(
-      'executionDidStart',
+    let response: GraphQLResponse;
+
+    // Ask plugins (eg, full query cache) for responses.
+    const pluginGraphQLResponses = (await dispatcher.invokeHookAsync(
+      'execute',
       requestContext as WithRequired<
         typeof requestContext,
         'document' | 'operation' | 'operationName'
       >,
-    );
+    )).filter(x => x);
+    if (pluginGraphQLResponses.length > 1) {
```

Comment:
I'm worried enforcing the rule that only one plugin can return a response will be too restrictive. I can even imagine wanting to add multiple caching plugins. An alternative would be to invoke hooks in order (instead of in parallel) until a plugin returns a response. (Maybe we could add a `Dispatcher.invokeHookSequentially`?)